src/registration.cpp

```diff
@@ -301,6 +350,34 @@ void RegistrationImpl::getPointXYZRGB (const Frame* undistorted, const Frame* re
   }
 }
 
+void Registration::getPointXYZ(const Frame *undistorted, int r, int c, float &x, float &y, float &z) const
+{
+    impl_->getPointXYZ(undistorted,r,c,x,y,z);
+}
+
```

Comment:
I do agree that we have to get rid of code duplication, but I prefer to do the other way around, which is to say implement getPointXYZRGB using getPointXYZ. I have updated my commit, have a look and let me know what you think!